cni/network/network.go

```diff
@@ -301,6 +295,35 @@ func addNatIPV6SubnetInfo(nwCfg *cni.NetworkConfig,
 	}
 }
 
+func acquireLockForStore(config *common.PluginConfig, plugin *netPlugin) error {
+	var err error
+	if err = plugin.Store.Lock(true); err != nil {
```

Comment:
following this err is hard. it might be better to make this a one-line declaration/initialization (`if err := ...; err != nil`) and in the nested `if`s below, explicitly return `err` or `nil` early instead of sometimes setting `err = nil` and returning `err` at the end of the function.

Reply:
i wanted to avoid multiple returns

Reply:
multiple returns are okay, in go it's encouraged to return early when you can. returning early on errors lets us keep the code linear and not need to nest it as deeply.